web/src/typeahead_helper.ts

```diff
@@ -650,14 +658,20 @@ export function sort_streams(matches: StreamPillData[], query: string): StreamPi
     return [...name_results.matches, ...desc_results.matches, ...desc_results.rest];
 }
 
-export function query_matches_person(query: string, person: User): boolean {
+export function query_matches_person(
+    query: string,
+    person: UserPillData | UserOrMentionPillData,
+): boolean {
     return (
         typeahead.query_matches_string_in_order(query, person.full_name, " ") ||
-        (Boolean(person.delivery_email) &&
+        (person.type === "user" &&
```

Comment:
This is a great improvement over using `Boolean`, yay.

Reply:
Okay this actually changes the logic a bit. Previously we were only matching when `delivery_email` was set, and now we always check for a match on email if it's not a mention. I think this is what we want? And it looks anecdotally like `delivery_email` is set consistently but I need to look into that more.

Reply:
I'll make this a commit after the main commit